src/main/java/io/lettuce/core/api/async/RedisJsonAsyncCommands.java

```diff
@@ -0,0 +1,300 @@
+     *
+     * @param arguments the {@link JsonMsetArgs} specifying the values to change.
+     * @return "OK" if the operation was successful, error otherwise
+     * @since 6.5
+     */
+    RedisFuture<String> jsonMSet(JsonMsetArgs... arguments);
+
+    /**
+     * Increment the number value stored at the specified {@link JsonPath} in the JSON document by the provided increment.
+     *
+     * @param key the key holding the JSON document.
+     * @param jsonPath the {@link JsonPath} pointing to the value to increment.
+     * @param number the increment value.
+     * @return a {@link List} of the new values after the increment.
+     * @since 6.5
+     */
+    RedisFuture<List<Number>> jsonNumincrby(K key, JsonPath jsonPath, Number number);
+
+    /**
+     * Return the keys in the JSON document that are referenced by the given {@link JsonPath}
+     *
+     * @param key the key holding the JSON document.
+     * @param jsonPath the {@link JsonPath} pointing to the value(s) whose key(s) we want.
+     * @return List<K> the keys in the JSON document that are referenced by the given {@link JsonPath}.
+     * @since 6.5
+     */
+    RedisFuture<List<K>> jsonObjkeys(K key, JsonPath jsonPath);
+
+    /**
+     * Report the number of keys in the JSON object at path in key
+     *
+     * @param key the key holding the JSON document.
+     * @param jsonPath the {@link JsonPath} pointing to the value(s) whose key(s) we want to count
+     * @return Long the number of keys in the JSON object at the specified path, or null if the path does not exist.
+     * @since 6.5
+     */
+    RedisFuture<List<Long>> jsonObjlen(K key, JsonPath jsonPath);
+
+    /**
+     * Sets the JSON value at a given {@link JsonPath} in the JSON document.
+     * <p>
+     * For new Redis keys the path must be the root. For existing keys, when the entire path exists, the value that it contains
+     * is replaced with the JSON value. For existing keys, when the path exists, except for the last element, a new child is
+     * added with the JSON value.
+     * <p>
+     * Adds a key (with its respective value) to a JSON Object (in a RedisJSON data type key) only if it is the last child in
+     * the path, or it is the parent of a new child being added in the path. Optional arguments NX and XX modify this behavior
+     * for both new RedisJSON data type keys and the JSON Object keys in them.
+     *
+     * @param key the key holding the JSON document.
+     * @param jsonPath the {@link JsonPath} pointing to the value(s) where we want to set the value.
+     * @param value the {@link JsonValue} to set.
+     * @param options the {@link JsonSetArgs} the options for setting the value.
+     * @return String "OK" if the set was successful, null if the {@link JsonSetArgs} conditions are not met.
+     * @since 6.5
+     */
+    RedisFuture<String> jsonSet(K key, JsonPath jsonPath, JsonValue<K, V> value, JsonSetArgs options);
```

Comment:
As mentioned in my other comment the overloads that accept default values are in the list of leftovers, so I will address that.

As for the String vs JsonValue question ... Using an object model (JsonValue) helps avoid decoding the data, in case the user does not want to do that. For example:

```java
        JsonPath sourcePath = JsonPath.of("..mountain_bikes[0:1].colors");
        JsonPath destinationPath = JsonPath.of("..mountain_bikes[1:2].colors");
        List<JsonValue<String, String>> value = redis.jsonGet(BIKES_INVENTORY, sourcePath);
        String result = redis.jsonSet(BIKES_INVENTORY, destinationPath, value.get(0), args);
```

Would result in the driver fetching the `colors` JSON array from the first bike in the list of bikes and setting it as the  list of colors for the second bike in the list. The whole operation is done with zero decoding or encoding (or parsing for that manner).

If the API was instead using String as an argument for `jsonSet` (or any other method) then we would have to decode the data from the wire (byte array -> string) and then encode it again.